chromium_src/third_party/blink/renderer/modules/speech/speech_synthesis.cc

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * you can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/modules/speech/speech_synthesis.h"
+#include "brave/third_party/blink/renderer/brave_farbling_constants.h"
+#include "third_party/blink/public/platform/web_content_settings_client.h"
+
+#define OnSetVoiceList OnSetVoiceList_ChromiumImpl
+#include "../../../../../../../third_party/blink/renderer/modules/speech/speech_synthesis.cc"
+#undef OnSetVoiceList
+
+namespace blink {
+
+void SpeechSynthesis::OnSetVoiceList(
+    Vector<mojom::blink::SpeechSynthesisVoicePtr> mojom_voices) {
+  voice_list_.clear();
+  BraveFarblingLevel farbling_level = BraveFarblingLevel::OFF;
+  if (ExecutionContext* context = GetExecutionContext()) {
+    if (blink::WebContentSettingsClient* settings =
+            brave::GetContentSettingsClientFor(context)) {
+      farbling_level = settings->GetBraveFarblingLevel();
+    }
+  }
+  if (farbling_level == BraveFarblingLevel::OFF) {
+    // farbling off -> call upstream function
+    OnSetVoiceList_ChromiumImpl(std::move(mojom_voices));
+    return;
+  }
+  if (farbling_level == BraveFarblingLevel::MAXIMUM) {
+    // maximum farbling -> return empty voice list
+    VoicesDidChange();
+    return;
+  }
+  mojom::blink::SpeechSynthesisVoicePtr fake_voice;
+  for (auto& mojom_voice : mojom_voices) {
+    if (!fake_voice && mojom_voice->is_default) {
+      // balanced farbling -> return real voices + one fake voice
+      if (ExecutionContext* context = GetExecutionContext()) {
+        fake_voice = mojom_voice.Clone();
+        fake_voice->is_default = false;
+        std::mt19937_64 prng = brave::BraveSessionCache::From(*context)
+                                   .MakePseudoRandomGenerator();
+        const char* kFakeNames[] = {
+            "Hubert", "Vernon", "Rudolph",   "Clayton",    "Irving",
+            "Wilson", "Alva",   "Harley",    "Beauregard", "Cleveland",
+            "Cecil",  "Reuben", "Sylvester", "Jasper"};
+        const int kFakeNamesCount = 14;
```

Comment:
`base::size` to the rescue! feel free to use it whenever you have a good ol' array.
https://github.com/chromium/chromium/blob/97.0.4690.7/base/cxx17_backports.h#L22-L27

Reply:
Logged https://github.com/brave/brave-browser/issues/19316 for followup